### PR TITLE
Implement TrigInterface class

### PR DESCRIPTION
**Pull request by jmamma**
@yatli 

Compiles, haven't uploaded yet.

I've only made the change to PageSelect page. The other pages will still attempt to use md_exploit.

### CODE REVIEW

File: avr/cores/megacommand/MCL/MCL.cpp [hunk 1]
```diff
@@ -43,8 +43,9 @@ void MCL::setup() {
 
   mcl_actions.setup();
   mcl_seq.setup();
-  MDSysexListener.setup(&Midi);
 
+  MDSysexListener.setup(&Midi);
+  trig_interface.setup(&Midi);
```
Comment (yatli) [inline avr/cores/megacommand/MCL/MCL.cpp:48]: Suggest:

Use 0x30 to query if TI mode is present.
Then multiplex md_exploit within trig interface?

Comment (jmamma) [reply avr/cores/megacommand/MCL/MCL.cpp:48]: Yeah that might be a good idea.

Would be even better if users upgrade their machines  :)

**Comment by yatli** [comment]
~~uploaded and not working.~~

trying 
eb97899

**Comment by jmamma** [comment]
It is sort of working. 

First trig press works. Subsequent actions dont update the trig number, I think the old sysex message is being "replayed"

**Comment by yatli** [comment]
The first press switches to the wrong page.
Misaligned payload?

**Comment by jmamma** [comment]
70   uint8_t trig = sysex->getByte(2);

trying.

**Comment by jmamma** [comment]
fixed.

**Comment by yatli** [comment]
can confirm, fixed.

**Comment by yatli** [comment]
ridiculous low latency.

**Comment by jmamma** [comment]
> ridiculous low latency.

Yeah, end_immediate() is why,

**Comment by jmamma** [comment]
So now that this is working. Tomorrow I will port the rest of the MCL code to use TI interface instead of md_exploit, with a possible fallback mechanism.

I think we need to have a few different TI modes.

Is it possible to hide the sequencer event LED flashes ?

**Comment by yatli** [comment]
wrt trig leds:

overlay (XOR), or override?

**Comment by yatli** [comment]
yeah, exactly this question.
I'll implement override now

**Comment by jmamma** [comment]
Override needed for step edit.

XOR for mixer page?

--

Do we get the MC to tell the MD what LEDs to light up, or maybe in some modes the MD can do that for us.

For example, step edit, we need to send the steps across (bitmask maybe?) 

But for mixer page let the MD work out which LEDs should be set.

**Comment by jmamma** [comment]
It's late here. i'm sure theres a simpler implementation to what i'm asking

**Comment by jmamma** [comment]
sleep time.

**Comment by yatli** [comment]
good night!
tmr I'll be catch a plane. It's Chinese Spring Festival here, off for a few days with no access to the gears.

**Comment by yatli** [comment]
> Do we get the MC to tell the MD what LEDs to light up, or maybe in some modes the MD can do that for us.

the logic can be implemented in MegaCommandTI directly.

**Comment by jmamma** [comment]
> good night!
> tmr I'll be catch a plane. It's Chinese Spring Festival here, off for a few days with no access to the gears.

No worries, enjoy your break.

File: avr/cores/megacommand/MCL/MCL.cpp (hunk 1 above)
Comment (jmamma) [reply avr/cores/megacommand/MCL/MCL.cpp:48]: I've decided to deprecate md_exploit.

Having both systems will add complexity to the code in which I do not want to manage. 

For example, there are some scenarios whereby the exploit must be turned off, during (specific sysex commands) and then turned back on again.

Comment (jmamma) [reply avr/cores/megacommand/MCL/MCL.cpp:48]: Also, it means we have to test both systems during fw upgrades. No thanks.

Comment (yatli) [reply avr/cores/megacommand/MCL/MCL.cpp:48]: I agree. Let's move forward.

Comment (yatli) [reply avr/cores/megacommand/MCL/MCL.cpp:48]: It also means that exploit-related bugs are all gone -- when you tweak the params, the TI no longer fails; no need to switch to track 16 anymore;

**Comment by jmamma** [comment]
```jmammarella@DESKTOP-DCRTMLQ:/mnt/c/Users/Justin/dev/mdhack/fw$ cat loader.s
        .text
        .align  2

        .globl  MegaCommandLoader

MegaCommandLoader:
  pea 0x002c0000
  pea 0x000e96e0    /* base image size = 0xe56e0; +offset 0x4000 */
  jsr 0x000001c4    /* UNPACK_FIRMWARE from e96e0 to 2c0000 */
  addq.l #0x8, %sp
  jsr 0x002c0000    /* MegaCommandInit (fw.text base address) */
  jmp 0x00213a0c    /* Resume MD_MAIN */
```

Mystery solved, MegaCommandInit address is hardcoded.

**Pull request merged**